server/events/vcs/github_client.go

```diff
@@ -307,6 +309,40 @@ func (g *GithubClient) PullIsMergeable(repo models.Repo, pull models.PullRequest
 	return true, nil
 }
 
+// Check if the Pull Request is locked with :lock emoji
+func (g *GithubClient) PullIsLocked(repo models.Repo, pull models.PullRequest) (bool, error) {
+	statuses, err := g.getRepoStatuses(repo, pull)
+	if err != nil {
+		return false, errors.Wrapf(err, "fetching repo statuses for repo: %s, and pull number: %d", repo.FullName, pull.Num)
+	}
+
+	for _, status := range statuses {
+		if status.GetContext() == SubmitQueueReadinessStatusContext {
```

Comment:
nit: readability can be impacted when you have something like:

```go

for .... {
  if .... {
      if ... {
      
          for ... {

           }

      }
  }

}

```


What i like to do is leverage `continue` to help with that ie.

```go
for ... {
   if ! .... {
      continue
   }

    // proceed with logic and save a tab

}

```

Reply:
Done!